mlrun/track/trackers/mlflow_tracker.py

```diff
@@ -0,0 +1,449 @@
+            client.set_tag(
+                run_id=run.info.run_id, key="mlrun-context-id", value=context.uid
+            )
+            client.set_tag(
+                run_id=run.info.run_id, key="mlrun-project-name", value=context.project
+            )
+
+            # Get the MLFlow run's tags and save them as labels:
+            context.set_label(key="mlflow-user", value=run.data.tags.get("mlflow.user"))
+            context.set_label(
+                key="mlflow-run-name", value=run.data.tags.get("mlflow.runName")
+            )
+            context.set_label(key="mlflow-run-id", value=run.info.run_id)
+            context.set_label(key="mlflow-experiment-id", value=run.info.experiment_id)
+
+            # Get the MLFlow run's parameters and save as job parameters:
+            context._parameters.update(run.data.params)
+
+            # Get the MLFlow run's metrics and log them as results:
+            results = run.data.metrics
+            context.log_results(results=results)
+
+            # Import the MLFlow run's artifacts to MLRun (model are logged after the rest of artifacts
+            # so the artifacts can be registered as extra data in the models):
+            artifacts = {}
+            model_paths = []
+            for artifact in client.list_artifacts(run_id=run.info.run_id):
+                # Get the artifact's local path (MLFlow suggests that if the artifact is already in the local filesystem
+                # its local path will be returned:
+                artifact_local_path = mlflow.artifacts.download_artifacts(
+                    run_id=run.info.run_id,
+                    artifact_path=artifact.path,
+                )
+                # Check if the artifact is a model (will be logged after the artifacts):
+                if artifact.is_dir and os.path.exists(
+                    os.path.join(artifact_local_path, "MLmodel")
+                ):
+                    model_paths.append(artifact_local_path)
+                else:
+                    # Log the artifact:
+                    artifact = MLFlowTracker._log_artifact(
+                        context_or_project=context,
+                        key=pathlib.Path(artifact.path).name.replace(".", "_"),
+                        local_path=artifact_local_path,
+                        tmp_path=tmp_dir,
+                    )
+                    artifacts[artifact.key] = artifact
+            if model_paths:
+                for model_path in model_paths:
+                    MLFlowTracker._log_model(
+                        context_or_project=context,
+                        model_uri=model_path,
+                        key=pathlib.Path(model_path).stem,
+                        metrics=results,
+                        extra_data=artifacts,
+                        tmp_path=tmp_dir,
+                    )
+
+    @staticmethod
+    def _log_model(
+        context_or_project: Union[MLClientCtx, MlrunProject],
```

Comment:
This assumes that the signatures are the same for both classes which may be true today and wrong tomorrow.
Why not have separate parameters? Since its a private method its not a big issue but I think its bad practice.

Reply:
you mean context and project?

Reply:
yes :)
```suggestion
        context: MLClientCtx,
        project: MlrunProject,
```